docs/proposals/opaque-integer-resources.md

```diff
@@ -0,0 +1,127 @@
+# Accounting and Consuming Opaque Integer Resources
+
+## Terms
+
+### Opaque node-level resource
+A resource associated with a node that has a consumable capacity, which is handled uniformly by the scheduler and Kubelet. (If there are specializations that hang off of the resource name, then it is not opaque!)
+
+## Requirements
+- Associate available opaque counted resources with existing nodes.
+- Request opaque resources for a pod.
+- The scheduler must account opaque resources similarly to existing first-class resources like logical CPU cores and main memory. Co-scheduled pods must not exceed the allocatable amount for any opaque resource.
+
+## Concrete use cases
+
+Even before implementing isolation, there is already user value in accounting opaque resources in the cluster.
+
+* Exclusive access to discrete co-processors (GPGPUs, MICs, FPGAs, etc.)
+* Counting slots allowed to access a shared parallel file system.
+* Support for tracking additional cgroup quotas like `cpu_rt_runtime`.
+
+## Goals
+1. Associate opaque integer resources with nodes.
+2. Request some portion of available capacity in the Pod spec during pod creation.
+3. Enable users to define and schedule on opaque resources in a vanilla Kubernetes cluster.
+
+## Non-goals
+1. Solve how opaque resource allocations are to be enforced/isolated.
+2. Solve how opaque resources are discovered, beyond existing APIs.
+3. Solve how cluster-level opaque resources should be advertised. 
```

Comment:
Yes. The advertising part is covered below. I will remove it from here. But we should make it clear we want to deal only with node-level resources for now. I will make appropriate changes to the document.